### ISSUE TITLE
Questions about training

### ISSUE THREAD

**Issue by LAW1223** [NVlabs/DiffusionNFT#37]
I want to express my sincere admiration for this work — it’s an incredibly insightful and well-executed project that has provided great value for my research on diffusion models.

During my training practice with the codebase, I’ve encountered several intriguing questions and hope to get some insights or suggestions from the team. The details are as follows:

**1. CFG > 1 slows down training convergence.** When sampling training samples, if the Classifier-Free Guidance (CFG) scale is set greater than 1, the convergence speed of the training process becomes significantly slower. I wonder if this is an expected phenomenon and what the underlying reasons are.

**2. Semantic knowledge learning with CFG = 1**
I am curious about the mechanism behind why the model can gradually learn semantic knowledge when sampling training samples with CFG = 1. What is the key factor that enables the model to capture semantic information under this setting?

**3. Choice between using v and x in loss calculation**
According to the formulas in the corresponding paper, the variable used for relevant calculations should be v. However, in the actual training code, x is adopted for loss computation. Although the theoretical principles of using v and x are similar, I want to ask:
Has the team tried using v for loss calculation in experiments?
Is the choice of x mainly due to the consideration of training stability? Or are there other factors such as computational efficiency or implementation convenience?

**4. Poor generalization performance of the trained model**
After the training is completed, the model only focuses on content related to the target object. When conducting generalization tests with other types of data, the generated results still only contain the target object with a white background. Are there any feasible solutions or optimization strategies to improve the generalization ability of the model?

**Comment by zhengkw18** [comment]
Thanks for your interest and attention to our work! We acknowledge that DiffusionNFT is a preliminary novel paradigm and may have limitations at its current stage. We appreciate all research efforts on DiffusionNFT. Regarding your questions:

1. We suspect that this is caused by the gap between the training samples and the policy to be optimized. As we only optimize the conditional model, the theoretically most correct way is also to use CFG=1 to collect training samples. The deeper mechanism behind this phenomenon is to be discovered.
2. We believe the contrast between positive and negative samples helps the model to capture semantic information. As noted, negative signals are essential in our method, and training with positive samples only will lead to collapse.
3. The transformation from v to x only results in a difference in the loss weighting between timesteps. As noted in our paper, we tried v + manual loss weightings, and found that the adaptive weighting consistently leads to faster reward increase. 
4. We recommend using multiple reward models to mitigate reward hacking to a single reward model.
